index.bs

```diff
@@ -2320,33 +2320,27 @@ The result of <dfn>parsing the blocklist</dfn> at a {{URL}} |url| is a [=list=]
 of {{USBBlocklistEntry}} objects produced by the following algorithm:
 
   1. Fetch |url| and let |contents| be its body, decoded as UTF-8.
-  1. Let |lines| be the result of invoking
-     {{String/split(separator, limit)}} on |contents| with separator
-     <code>'\n'</code>.
+  1. Let |lines| be the result of [=strictly splitting=] |contents| starting
+     from the beginning of |contents| on code point <code>'\n'</code>.
   1. Let |blocklist| be an empty [=list=].
   1. [=list/For each=] |line| of |lines|:
-      1. Let |commentBegin| be the result of invoking
-         |line|.{{String/indexOf()}} with <code>'#'</code>.
-      1. If |commentBegin| is not -1, set |line| to the result of invoking
-         |line|.{{String/substring()}} with 0 and |commentBegin|.
-      1. Set |line| to the result of |line|.{{String/trimEnd()}}.
-      1. Let |components| be the result of invoking
-         |line|.{{String/split(separator, limit)}} with separator
-         <code>':'</code>.
+      1. Set |line| to the result of [=collecting a sequence of code points=]
+         not equal to <code>'#'</code> from |line| starting from the beginning
+         of |line|.
+      1. Set |line| to the result of [=stripping leading and trailing ASCII
+         whitespace=] from |line|.
+      1. Let |components| be the result of [=strictly splitting=] |line|
+         starting from the beginning of |line| on code point <code>':'</code>.
       1. If the [=list/size=] of |components| is not 2 or 3,
          [=iteration/continue=].
-      1. Let |idVendor| be the result of invoking
-         {{Number}}.{{Number/parseInt(string, radix)}} with |components|[0]
-         and 16, or [=iteration/continue=] if a {{TypeError}} is thrown.
-      1. Let |idProduct| be the result of invoking
-         {{Number}}.{{Number/parseInt(string, radix)}} with |components|[1]
-         and 16, or [=iteration/continue=] if a {{TypeError}} is thrown.
+      1. Let |idVendor| be the result of interpreting |components|[0] as a
+         hexadecimal number.
+      1. Let |idProduct| be the result of interpreting |components|[1] as a
+         hexadecimal number.
       1. If the [=list/size=] of |components| is:
           * 2: Let |bcdDevice| be <code>0xFFFF</code>.
```

Comment:
This line is no longer necessary.

Reply:
I think this is still needed, otherwise `bcdDevice` will be uninitialized when we create the `USBBlocklistEntry`.

Reply:
Ah, then this needs to be swapped with the "if" above so that it is set to this default when components only has 2 elements.

Reply:
I don't understand the issue. If we move the `if` lower then `|components|[1]` will be an OOB access if the line doesn't have any ':' code points.

How about:

```
      1. Let |components| be the result of [=strictly splitting=] |line|
         starting from the beginning of |line| on code point <code>':'</code>.
      1. If the [=list/size=] of |components| is not 2 or 3,
         [=iteration/continue=].
      1. Let |idVendor| be the result of interpreting |components|[0] as a
         hexadecimal number.
      1. Let |idProduct| be the result of interpreting |components|[1] as a
         hexadecimal number.
      1. Let |bcdDevice| be <code>0xFFFF</code>.
      1. If the [=list/size=] of |components| is 3, set |bcdDevice| to the
         result of interpreting |components|[2] as a hexadecimal number.
      1. [=list/Append=] a new {{USBBlocklistEntry}} with |idVendor|,
         |idProduct|, and |bcdDevice| to |blocklist|.
```

Reply:
That LGTM.